src/ray/core_worker/transport/direct_actor_transport.h

```diff
@@ -144,19 +165,9 @@ class CoreWorkerDirectActorTaskSubmitter {
   absl::flat_hash_map<ActorID, rpc::KillActorRequest> pending_force_kills_
       GUARDED_BY(mu_);
 
-  /// Map from actor id to the actor's pending requests. Each actor's requests
-  /// are ordered by the task number in the request.
-  absl::flat_hash_map<ActorID, std::map<int64_t, std::unique_ptr<rpc::PushTaskRequest>>>
-      pending_requests_ GUARDED_BY(mu_);
-
-  /// Map from actor id to the send position of the next task to queue for send
-  /// for that actor. This is always greater than or equal to next_send_position_.
-  absl::flat_hash_map<ActorID, int64_t> next_send_position_to_assign_ GUARDED_BY(mu_);
-
-  /// Map from actor id to the send position of the next task to send to that actor.
-  /// Note that this differs from the PushTaskRequest's sequence number in that it
-  /// increases monotonically in this process independently of CallerId changes.
-  absl::flat_hash_map<ActorID, int64_t> next_send_position_ GUARDED_BY(mu_);
+  /// Map from actor id to the actor's pending requests from this worker. Each
+  /// actor's requests are ordered by the task number in the request.
+  absl::flat_hash_map<ActorID, ClientQueue> task_queues_ GUARDED_BY(mu_);
```

Comment:
Should this be moved into the actor manager class?

Reply:
Yeah, actually I had a version of this that had all the task queues in the actor manager, but it ended up being confusing because the actor task is actually not in any queue while its dependencies are being resolved...I think there is actually a bug in the current implementation where an actor task could hang if its actor dies as the task is being resolved.

I'll think about whether we can just move the queue completely into the actor manager in a better way.